### PR TITLE
fix: correct Danger source test coverage check

### PR DESCRIPTION
**Pull request by rajzik**
### Summary
- Fix `checkSourceFilesHaveTests` so it reports source files only when no matching test file changed.
- Add a TypeScript Vitest regression suite covering matching-test, missing-test, and snapshot-file edge cases.
- Document the repo testing preference in `AGENTS.md`: use Vitest and TypeScript for tests, not JavaScript test files.
- Add a patch changeset for `@rajzik/danger-configuration`.

### Testing
- `pnpm --filter @rajzik/danger-configuration lint`
- `pnpm --filter @rajzik/danger-configuration test`
- `pnpm --filter @rajzik/danger-configuration build`
- `pnpm typecheck`
- `pnpm test`

<div><a href="https://cursor.com/agents/bc-bc91893c-3252-4fa4-9483-82ab3a1093dc"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-bc91893c-3252-4fa4-9483-82ab3a1093dc"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

## Release Notes

* **Bug Fixes**
  * Fixed the strict source test coverage check to correctly identify and report source files missing matching tests.

* **Chores**
  * Updated testing standards: all newly added or modified tests must be written using Vitest with TypeScript.
  * Added project-level test infrastructure and configuration.

### CODE REVIEW

File: packages/config-danger/tests/code.test.mjs [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import assert from 'node:assert/strict';
```
Comment (rajzik) [inline packages/config-danger/tests/code.test.mjs]: Use vitest and typescript
